### PR TITLE
br: retry when meet hole in key space

### LINKED ISSUE #27426: ERROR 8125 (HY000): Restore failed: last region's endKey < startKey
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

3 tidb, 3 pd, 3 tikv

```
// backup many list partition tables
RESTORE DATABASE * FROM "local:///tmp/lp_backup/";
```

### 2. What did you expect to see? (Required)
restore success

### 3. What did you see instead (Required)
```
mysql> RESTORE DATABASE * FROM "local:///tmp/lp_backup/";
ERROR 8125 (HY000): Restore failed: region endKey not equal to next region startKey, endKey: 748000000000000dffea00000000000000f8, startKey: 748000000000000dffeb00000000000000f8: [BR:PD:ErrPDBatchScanRegion]batch scan region
```
### 4. What is your TiDB version? (Required)

```
Release Version: v5.2.0
Edition: Community
Git Commit Hash: d4d667fba5e82f108c62cd690fcf73c685d7612f
Git Branch: heads/refs/tags/v5.2.0
UTC Build Time: 2021-08-18 12:35:12
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #27426 

Problem Summary:
When scanning region, there may be some hole in key space by design of PD. We could retry in this condition. 

### What is changed and how it works?

What's Changed:
Retry when meet `ErrPDBatchScanRegion`

### Check List 

Tests 

- [ ] Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a bug caused restore fail due to `ErrPDBatchScanRegion`.
```

### CODE REVIEW

File: br/pkg/restore/split.go [hunk 1]
```diff
@@ -112,12 +112,13 @@ SplitRegions:
 	for i := 0; i < SplitRetryTimes; i++ {
 		regions, errScan := PaginateScanRegion(ctx, rs.client, minKey, maxKey, ScanRegionPaginationLimit)
 		if errScan != nil {
+			if berrors.ErrPDBatchScanRegion.Equal(errScan) {
+				log.Warn("inconsistent region info get.", logutil.ShortError(errScan))
```
Comment (kennytm) [inline br/pkg/restore/split.go:116]: include the minKey and maxKey into the log? the (short) error message according to #27426 seems to be only "batch scan region".

Comment (YuJuncen) [reply br/pkg/restore/split.go:116]: OK, let's redact it then.

**Review by glorv — APPROVED** [review]

**Comment by YuJuncen** [comment]
image

Unit test shows the retry has been triggered.

**Review by kennytm — APPROVED** [review]

**Comment by YuJuncen** [comment]
```
[2021-08-23T01:58:59.706Z] WARNING: DATA RACE
[2021-08-23T01:58:59.706Z] Read at 0x000005e1e900 by goroutine 56:
[2021-08-23T01:58:59.706Z]   time.Now()
[2021-08-23T01:58:59.706Z]       /usr/local/go/src/time/time.go:1073 +0xf6
[2021-08-23T01:58:59.706Z]   time.sendTime()
[2021-08-23T01:58:59.706Z]       /usr/local/go/src/time/sleep.go:149 +0x49
[2021-08-23T01:58:59.706Z] 
[2021-08-23T01:58:59.706Z] Previous write at 0x000005e1e900 by goroutine 69:
[2021-08-23T01:58:59.706Z]   ??()
[2021-08-23T01:58:59.706Z]       -:0 +0xffffffffffffffff
[2021-08-23T01:58:59.706Z]   fmt.Sprintf()
[2021-08-23T01:58:59.706Z]       /usr/local/go/src/fmt/print.go:221 +0xc9
[2021-08-23T01:58:59.706Z]   github.com/pingcap/tidb/br/pkg/lightning/mydump.setDatumByInt()
```
🤔
/run-unit-test

**Comment by YuJuncen** [comment]
/run-unit-test

executor failed.

**Pull request merged**